Stripe/PublicHeaders/STPAPIClient.h

```diff
@@ -127,6 +127,19 @@ static NSString *const STPSDKVersion = @"11.5.0";
 
 @end
 
+#pragma mark Connect Accounts
+
+/**
+ Stripe extensions for working with Connect Accounts
+ */
+@interface STPAPIClient (ConnectAccounts)
+
+- (void)createTokenWithConnectAccount:(id)obj completion:(__nullable STPTokenCompletionBlock)completion;
```

Comment:
obj needs to turn into:
```
account: {
  legal_entity: {...},
  tos_shown_and_accepted: true,
}
```
with either `legal_entity` **or** `tos_shown_and_accepted`, or both.

I don't know whether I want to create another `...Params` class, or change this method to take two nullable parameters.

I also think it's unfortunate that `Bool?` is `nullable NSNumber` in Obj-C

Reply:
Hmm, I think it'd be better to make this method take parameters rather than make a parent `AccountParams` class, but don't feel strongly.

Is it necessary to model `tos_shown_and_accepted` as a `Bool?`, or can we treat `tos_shown_and_accepted: nil` as `tos_shown_and_accepted: false`?

Reply:
I don't think that would work. Valid values for `tos_shown_and_accepted` are `{true, nil}`. Some basic use cases:

* `account: { tos_shown_and_accepted: true }`: minimum required to set up a new account
* `account: { legal_entity: { ... }, tos_shown_and_accepted: true }`: setting up new account with some data about the legal entity
* `account: { legal_entity: { ... } }`: valid for updating a pre-existing account
* `account: { tos_shown_and_accepted: false, ... }`: rejected by the server as invalid

As I understand it, the initial purpose for this API is collecting positive affirmation that the user has seen & accepted the ToS, instead of the old world where the platform is allowed to vouch that fact as true. However, it's *also* just as usable for updating the legal entity info directly.


I'll think about `AccountParams` a little more. I like that it's easier to add new fields in the future, and the possibility of providing >1 constructors to help understand the `legal_entity/tos_shown_and_accepted` distinction. But it's a lot just for two optional fields

Reply:
ah interesting, if that's the case then perhaps an `AccountParams` object with multiple constructors makes sense. thanks for the explanation!